### PR TITLE
Update Supplier.php

### PR DESCRIPTION
**Pull request by switch-cheehau**
add supplier.addressformat as configuration

refer to #239
Sorry, i've screwed up the commits in previous PR. re-create this PR for clean code.

### CODE REVIEW

File: lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php [hunk 1]
```diff
@@ -179,7 +191,7 @@ public function setConfigFE( \Aimeos\MShop\Order\Item\Base\Service\Iface $orderS
 		if( ( $code = $attributes['supplier.code'] ) != '' )
 		{
 			// add short address as attribute for summary page / customer email
-			$attributes['supplier.address'] = $this->feConfig['supplier.code']['short'][$code];
+			$attributes['supplier.address'] = $this->feConfig['supplier.code'][$this->getConfigValue('supplier.addressformat')][$code];
```
Comment (aimeos) [inline lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php:194]: You should use a default value if it's not set like:
```php
$format = $this->getConfigValue( 'supplier.addressformat', 'short' );
$attributes['supplier.address'] = $this->feConfig['supplier.code'][$format][$code];
```

Comment (switch-cheehau) [reply lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php:194]: Can i do it in oneline?
```
$attributes['supplier.address'] = $this->feConfig['supplier.code'][$this->getConfigValue('supplier.addressformat','short')][$code];
```

Comment (aimeos) [reply lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php]: Yes but the line will be very long and we try to keep the line length below 120 chars.

Comment (switch-cheehau) [reply lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php]: ah, okay, understand, i'll change.

Comment (switch-cheehau) [reply lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php]: done

**Comment by aimeos** [comment]
We've discussed this a bit internally and came to the conclusion that adding just another format may not fit the requirements of many users as they may have different wishes for the address format depending on the country and/or intended use.

Our recommendation would be to add the format string directly to the configuration instead of a fixed list of format options.

**Comment by switch-cheehau** [comment]
> We've discussed this a bit internally and came to the conclusion that adding just another format may not fit the requirements of many users as they may have different wishes for the address format depending on the country and/or intended use.
> 
> Our recommendation would be to add the format string directly to the configuration instead of a fixed list of format options.

make sense. 
If you need my help to continue the PR,
do you have any existing example that i can refer to implement the changes?

**Comment by aimeos** [comment]
Yes, your help would be valuable :-)
The solution is to use that string from provider configuration: https://github.com/aimeos/aimeos-core/blob/master/lib/mshoplib/src/MShop/Service/Provider/Decorator/Supplier.php#L72-L82

**Comment by switch-cheehau** [comment]
create new PR for different approach.

**Pull request closed without merging**